src/ray/object_manager/push_manager.h

```diff
@@ -139,7 +146,10 @@ class PushManager {
   int64_t chunks_remaining_ = 0;
 
   /// Tracks all pushes with chunk transfers in flight.
-  absl::flat_hash_map<PushID, std::unique_ptr<PushState>> push_info_;
+  absl::flat_hash_map<PushID, std::shared_ptr<PushState>> push_info_;
+
+  /// The list of push requests with chunks waiting to be sent.
+  std::list<std::pair<PushID, std::shared_ptr<PushState>>> pending_push_requests_;
```

Comment:
Let's make it unique_ptr in `push_info_` and here we can just do `PushState*`: this makes the ownership clear and we know for all the push requests in `pending_push_requests_` they must also be in `push_info_`

nit: `pending_push_requests_` -> `push_requests_with_chunks_to_send_`